src/fenwick.jl

```diff
@@ -0,0 +1,114 @@
+"""
+    FenwickTree{T}(n)
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) of length `n`.
+ 
+"""
+mutable struct FenwickTree{T}
+    bi_tree::Vector{T} #bi_tree is shorthand for Binary Indexed Tree, an alternative name for Fenwick Tree
+    n::Integer
+end
+
+FenwickTree{T}() where T = FenwickTree{T}(0)
+FenwickTree{T}(n::Integer) where T = FenwickTree{T}(zeros(T, n), n)
+
+"""
+    FenwickTree(arr::AbstractArray) 
+    
+Constructs a [`FenwickTree`](https://en.wikipedia.org/wiki/Fenwick_tree) from `arr`.
+ 
+"""
+function FenwickTree(a::AbstractVector{U}) where U
+    n = size(a)[1]
+    tree = FenwickTree{U}(n)
+    @inbounds for i = 1:n
+        inc!(tree, i, a[i])
+    end
+    tree
+end
+
+length(ft::FenwickTree{T}) where T = ft.n
+
+"""
+    inc!(ft::FenwickTree{T}, ind, val)
+
+Increases the value of the [`FenwickTree`] by `val` from the index `ind` upto the length of the Fenwick Tree.
+
+""" 
+function inc!(ft::FenwickTree{T}, ind::Int, val) where T
```

Comment:
Is it reasonable for `val` to default to `1` here?  Or is it more typical to increment an index by an arbitrary amount?

Reply:
Sure, any default value would suffice, `1` seems pretty reasonable.